cgroup.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"github.com/docker/docker/pkg/parsers"
+)
+
+// set function in variable to overwrite for testing.
+var getCpusetGuest = func() string {
+	cpusetGuestByte, err := ioutil.ReadFile("/sys/fs/cgroup/cpuset/cpuset.cpus")
+	if err != nil {
+		fmt.Println("Error:", err)
+	}
+
+	return string(cpusetGuestByte)
+}
+
+// Return the best match for cpuset list in the guest.
+// The runtime caller may apply cpuset for specific CPUs in the host.
+// The CPUs may not exist on the guest as they are hotplugged based
+// on cpu and qouta.
+// This function return a working cpuset to apply on the guest.
+func getAvailableCpusetList(cpusetReq string) (string, error) {
+
+	cpusetGuest := getCpusetGuest()
+
+	cpusetListReq, err := parsers.ParseUintList(cpusetReq)
+	if err != nil {
+		return "", err
+	}
+
+	cpusetGuestList, err := parsers.ParseUintList(strings.TrimSpace(string(cpusetGuest)))
```

Comment:
cpusetGuest is a string, why is `string()` needed?

Reply:
Good catch, I did not removed when factored out `getCpusetGuest()`